src/client/_converter.py

```diff
@@ -11,21 +11,21 @@
 from ..util._extendable_enum import ExtendableStrEnum
 
 _CATTRS_VERSION = tuple(map(int, pkg_resources.get_distribution('cattrs').version.split('.')))
+
 if _CATTRS_VERSION < (22, 2, 0):
     converter = cattr.Converter()
 else:
     converter = cattr.converters.BaseConverter()
 
 converter.register_structure_hook_func(
     lambda type_: hasattr(type_, 'structure'),
-    lambda data, type_: type_.structure(data)  # type: ignore
+    lambda data, type_: getattr(type_, 'structure').__func__(type_, data)  # type: ignore
```

Comment:
Why this is required?

Reply:
I was not looking into details, but I think classmethod decorator breaks smth, idk.
Example:
```python
from typing import Generic, TypeVar


T = TypeVar('T')


class A(Generic[T]):
    data: T

    @classmethod
    def structure(cls, data):
        print(cls)
        print(data)


if __name__ == '__main__':
    data = {'data': 1}
    A[int].structure(data)  # printed: <class '__main__.A'>
    getattr(A, 'structure').__func__(A[int], data)  # printed __main__.A[int]
```